lib/schema.js

```diff
@@ -0,0 +1,139 @@
+const _ = require('lodash')
+const Joi = require('@hapi/joi')
+const { SORT_BY, SORT_DIRECTIONS } = require('./sort-pull-requests')
+
+const DEFAULT_CONFIG = Object.freeze({
+  'name-template': '',
+  'tag-template': '',
+  'change-template': `* $TITLE (#$NUMBER) @$AUTHOR`,
+  'no-changes-template': `* No changes`,
+  'version-template': `$MAJOR.$MINOR.$PATCH`,
+  categories: [],
+  'exclude-labels': [],
+  replacers: [],
+  'sort-by': SORT_BY.mergedAt,
+  'sort-direction': SORT_DIRECTIONS.descending
+})
+
+module.exports.DEFAULT_CONFIG = DEFAULT_CONFIG
+
+const category = Joi.object()
+  .keys({
+    title: Joi.string().required(),
+    label: Joi.array()
+      .items(Joi.string())
+      .single(),
+    labels: Joi.array()
+      .items(Joi.string())
+      .single()
+  })
+  .xor('label', 'labels')
+
+const searchReplace = Joi.object().keys({
+  search: Joi.string()
+    .required()
+    .error(new Error('"search" is required and must be a regexp or a string'))
+    .description(
+      'The text to search for in a regex or a string representation'
+    ),
+  replace: Joi.string()
+    .allow('')
+    .required()
+    .description(
+      'Replacement of text, it is possible to reference regex match groups by "$number" like $2 for second match group'
+    )
+})
+
+const fields = {
+  branches: Joi.array()
+    .items(Joi.string())
+    .single()
+    .description(
```

Comment:
I've been thinking about the descriptions in the schema. While they do add value, I foresee them getting out of sync with the docs. For now I'd like to request that we remove all descriptions, the validation and JSON schema alone should be a big improvement :)

Reply:
sure thing, not a problem for me 👍